state/azure/cosmosdb/cosmosdb.go

```diff
@@ -685,14 +683,16 @@ func populatePartitionMetadata(key string, requestMetadata map[string]string) st
 }
 
 func isNotFoundError(err error) bool {
+	fmt.Printf("Check isNotFoundError: %v\n", err)
 	if err == nil {
 		return false
 	}
 
 	if requestError, ok := err.(*azcore.ResponseError); ok {
-		if requestError.ErrorCode == statusNotFound {
+		if requestError.StatusCode == 404 {
```

Comment:
Linter calls this line out as well to replace 404 with http.StatusNotFound.

It's better to wait for a review and approval..